entrypoint.sh

```diff
@@ -100,37 +99,139 @@ function super_lint() {
   /action/lib/linter.sh
 }
 
+function get_files_matching_filetype() {
+  local filetype=$1
+  shift
+
+  matching_files=()
+
+  for file in "$@"; do
+    filename=$(basename "$file")
+    
+    if [[ "$filename" == *"$filetype" ]]; then
+      matching_files+=("$file")
+    fi
+  done
+  echo "${matching_files[@]}"
+}
+
+function linter_failed() {
+  return=$1
+  shift 
+  name=$1
+
+  if [[ "${return:-1}" != 0 ]]; then
+    linter_exit_codes+=(["$name"]="${return}")
+  else
+    return "${return}"
+  fi
+}
+
+function lint_files() {
+  exec &>> "${linter[logfile]}"
+  if [[ "${linter[filetype]}" = "all" ]]; then
+    cmd="${linter[name]} ${linter[args]}"
+    eval "$cmd"
+  else
+    for file in $(get_files_matching_filetype "${linter[filetype]}" "${included[@]}"); do 
+      if [[ "${linter[executor]+x}" ]]; then
+        cmd="${linter[executor]} -c \"${linter[name]} ${linter[args]} ${file}\""
+      else
+        cmd="${linter[name]} ${linter[args]} ${file}"
+      fi
+      eval "$cmd"
+    done
+  fi
+}
+
 function seiso_lint() {
+  echo -e "\nRunning Seiso Linter\n--------------------------\n"
+
   excluded=()
   included=()
 
   while read -r file; do
-    # Apply filter with =~ to ensure it is aligned with github/super-linter
     if [[ -n ${INPUT_EXCLUDE:+x} && "${file}" =~ ${INPUT_EXCLUDE} ]]; then
       excluded+=("${file}")
       continue
     fi
-
     included+=("${file}")
+  done < <(find . -path "./.git" -prune -or -type f)
+  
+  declare -gA linter_exit_codes
+  declare -gA pids
 
-    # Check Dockerfiles
-    if [[ "${file}" = *Dockerfile ]]; then
-      dockerfile_lint -f "${file}" -r /etc/opt/goat/oci.yml
-    fi
+  input="/etc/opt/goat/linters.json"
 
-    # Check .md file spelling and links
-    if [[ "${file}" = *.md ]]; then
-      npx cspell -c /etc/opt/goat/cspell.config.js -- "${file}"
-      npx markdown-link-check --config /etc/opt/goat/links.json --verbose "${file}"
+  while read -r line; do
+    if [[ $line == \#* ]]; then
+      continue
     fi
-  done < <(find . -path "./.git" -prune -or -type f)
-
-  echo "Scanned ${#included[@]} files"
-  echo "Excluded ${#excluded[@]} files"
+    
+    unset linter
+    declare -gA linter
+ 
+    while IFS='=' read -r key value; do
+      value=$(echo "$value" | tr -d "'")
+      linter["$key"]=$value
+    done < <(echo "$line" | jq -r 'to_entries|map("\(.key)=\(.value|tostring)")|.[]')
+        
+    linter+=([logfile]="/etc/opt/goat/logs/${linter[name]}.log")
+
+    echo "===============================" >> "${linter[logfile]}"
+    echo "Running linter: ${linter[name]}"
+    echo "${linter[name]^^}" >>"${linter[logfile]}"
+
+    lint_files & 
+    pid=$!
+    pids["$pid"]+="${linter[name]}"
```

Comment:
Why are we `+=`ing here? From what I can see, `pids` is a global associative array, and we're appending a string to an empty string which is the default value. We can just do a `=` assignment.
```bash
jonzeolla@~$ echo ${thearray[c]}

jonzeolla@~$ thearray[c]+='example'
jonzeolla@~$ echo ${thearray[c]}
example
```